src/categorical_algebra/DataMigrations.jl

```diff
@@ -41,102 +41,107 @@ a "duc query" (disjoint union of conjunctive queries).
 """
 const GlucQuery{C<:FinCat} = Diagram{id,<:TypeCat{<:Diagram{op,C}}}
 
-""" Functor defining a pullback or delta data migration.
+""" A data migration is guaranteed to contain a functor between
+schemas that can be used to migrate data between acsets
+on those schemas. This functor ``F`` may be partial, in which case
+it contains extra data describing how to use it to execute a migration. The
+migration may be a pullback data migration ([`DeltaMigration`](@ref)), specified by a
+functor ``D → C`` between the schemas, or a conjunctive, gluing,
+duc, or gluc (contravariant) data migration. Alternatively, ``F`` may
+be a schema functor specifying a ``Σ`` migration in the covariant direction.
 """
-const DeltaSchemaMigration{D<:FinCat,C<:FinCat} = FinFunctor{D,C}
+abstract type AbstractDataMigration{F<:FinDomFunctor} end
+func(M::AbstractDataMigration) = M.functor
```

Comment:
I'd strongly prefer to rename `func` to `functor` because:

- `func` could mean function or functor
- `functor` is only two characters longer
- `functor` is already a generic function that Catlab exports, so won't have to add another, inconsistent name to the namespace

You'll need `import ...Syntax: functor` at the top of the file.